contracts/LinearCodeAddressGenerator.cdc

```diff
@@ -0,0 +1,112 @@
+
+/// LinearCodeAddressGenerator allows generating and validating Flow account addresses.
+access(all)
+contract LinearCodeAddressGenerator {
+
+    /// The code word for addresses on Flow Mainnet.
+    access(all)
+    let codeWordMainnet: UInt64
+
+    /// The code word for addresses on Flow Testnet.
+    access(all)
+    let codeWordTestnet: UInt64
+
+    /// The code word for addresses on transient networks, like the Flow Emulator.
+    access(all)
+    let codeWordTransient: UInt64
+
+    /// Rows of the generator matrix G of the [64,45]-code used for Flow addresses.
+    /// G is a (k x n) matrix with coefficients in GF(2), each row is converted into
+    /// a big endian integer representation of the GF(2) raw vector.
+    /// G is used to generate the account addresses
+    access(all)
+    let generatorMatrixRows: [UInt64; 45]
+
+    /// Columns of the parity-check matrix H of the [64,45]-code used for Flow addresses.
+    /// H is a (n x p) matrix with coefficients in GF(2), each column is converted into
+    /// a big endian integer representation of the GF(2) column vector.
+    /// H is used to verify a code word is a valid account address.
+    access(all)
+    let parityCheckMatrixColumns: [UInt64; 64]
+
+    init() {
```

Comment:
I see that the `LinearAddressGenerator` in the reference implementation has an [`index` field](https://github.com/onflow/flow-go/blob/master/model/flow/address.go#L145). I'm assuming the field isn't necessary here since the the `index` state isn't tracked at the execution level, but am just making sure.

Reply:
The Cadence `LinearAddressGenerator` just implements the index <> address code, i.e. is more a port of of just [`linearCodeImpl`](https://github.com/onflow/flow-go/blob/8a3055c826a60f7a71c7a829d5d9e967c04940b0/model/flow/chain.go#L128). 

`linearCodeAddressGenerator` in the Go code is a stateful sequence/iterator, which isn't needed, but we could add a type for it to the contract once needed.